formats/gltf/writer.go

```diff
@@ -339,88 +357,175 @@ func rgbToFloatArr(c color.Color) [3]float64 {
+				rotations = append(rotations, group.instances[i].trs.Rotation().Vector4())
+				scales = append(scales, group.instances[i].trs.Scale())
 			}
-			w.extensionsUsed[extGpuInstancingID] = true
+		}
+
+		// Create instancing attributes
+		instances := ExtGpuInstancing{
+			Attributes: make(map[string]int),
+		}
+
+		instances.Attributes["TRANSLATION"] = len(w.accessors)
+		w.WriteVector3(AccessorComponentType_FLOAT, iter.Array(positions))
+
+		instances.Attributes["SCALE"] = len(w.accessors)
+		w.WriteVector3(AccessorComponentType_FLOAT, iter.Array(scales))
 
-			instances := ExtGpuInstancing{
-				Attributes: make(map[string]int),
+		instances.Attributes["ROTATION"] = len(w.accessors)
+		w.WriteVector4(AccessorComponentType_FLOAT, iter.Array(rotations))
+
+		newNode.Extensions[extGpuInstancingID] = instances
+
+		w.nodes = append(w.nodes, newNode)
+		w.scene = append(w.scene, nodeIndex)
+
+		if group.isAnimated() {
+			skinNode := nodeIndex
+			if group.skeleton != nil {
+				var skinIndex *int
+				skinIndex, skinNode = w.AddSkin(*group.skeleton)
+				w.nodes[nodeIndex].Skin = skinIndex
 			}
 
-			positions := make([]vector3.Float64, len(model.GpuInstances))
-			rotations := make([]vector4.Float64, len(model.GpuInstances))
-			scales := make([]vector3.Float64, len(model.GpuInstances))
-			for i, t := range model.GpuInstances {
-				positions[i] = t.Position()
-				rotations[i] = t.Rotation().Vector4()
-				scales[i] = t.Scale()
+			if len(group.animations) > 0 {
+				w.AddAnimations(group.animations, *group.skeleton, skinNode)
 			}
+		}
+
+		return nil
+	}
+
+	if group.isAnimated() && len(group.instances) > 1 {
+		// This really has already been checked earlier and should never happen.
+		// This limitation can be lifted once node children mechanics is implemented.
+		return fmt.Errorf("animated model %q has multiple instances, but GPU instancing is disabled", group.instances[0].name)
+	}
 
-			instances.Attributes["TRANSLATION"] = len(w.accessors)
-			w.WriteVector3(AccessorComponentType_FLOAT, iter.Array(positions))
+	// Create individual nodes for each instance
+	for _, instance := range group.instances {
+		nodeIndex := len(w.nodes)
+		newNode := Node{
+			Mesh: &group.meshIndex,
+			Name: instance.name,
+		}
 
-			instances.Attributes["SCALE"] = len(w.accessors)
-			w.WriteVector3(AccessorComponentType_FLOAT, iter.Array(scales))
+		// Set transformations
+		if instance.trs != nil {
+			posArr := instance.trs.Position().ToFixedArr()
+			scaleArr := instance.trs.Scale().ToFixedArr()
 
-			instances.Attributes["ROTATION"] = len(w.accessors)
-			w.WriteVector4(AccessorComponentType_FLOAT, iter.Array(rotations))
+			// Only set rotation if it's not identity quaternion
+			rot := instance.trs.Rotation()
+			if rot.W() != 1 || rot.Dir().X() != 0 || rot.Dir().Y() != 0 || rot.Dir().Z() != 0 {
+				rotArr := rot.ToArr()
+				newNode.Rotation = &rotArr
+			}
 
-			newNode.Extensions[extGpuInstancingID] = instances
+			newNode.Translation = &posArr
```

Comment:
could we do same cheks here like we do the rotation? only set position if it's not (0,0,0), and only set scale if it's not (1,1,1)